src/pages/CommandBlocklistView/CommandBlocklistView.tsx

```diff
@@ -0,0 +1,132 @@
+
+  const commandListData = useMemo(() => {
+    const handleClick = (command: CommandRow) => {
+      const selectedIndex = selectedCommands.indexOf(command)
+      let newSelected: CommandRow[] = []
+
+      if (selectedIndex === -1) {
+        newSelected = newSelected.concat(selectedCommands, command)
+      } else if (selectedIndex === 0) {
+        newSelected = newSelected.concat(selectedCommands.slice(1))
+      } else if (selectedIndex === selectedCommands.length - 1) {
+        newSelected = newSelected.concat(selectedCommands.slice(0, -1))
+      } else if (selectedIndex > 0) {
+        newSelected = newSelected.concat(
+          selectedCommands.slice(0, selectedIndex),
+          selectedCommands.slice(selectedIndex + 1),
+        )
+      }
+
+      setSelected(newSelected)
+    }
+    return commands.map((command: CommandRow): CommandRow => {
+      return {
+        namespace: command.namespace,
+        system: command.system,
+        command: command.command,
+        name: command.name,
+        action: (
+          <Checkbox
+            checked={selectedCommands.indexOf(command) > -1}
+            onClick={() => handleClick(command)}
+          />
+        ),
+      }
+    })
+  }, [commands, selectedCommands])
+
+  const blocklistData = useMemo(() => {
+    const delCommand = (id: string | undefined) => {
+      if (id) {
+        deleteBlockList(id)
+      } else {
+        console.error('Command had no ID - cannot delete')
+      } // TODO: search for command + system + namespace combo?
+    }
+
+    return (
+      blocked?.map((command: CommandBase): CommandRow => {
+        return {
+          namespace: command.namespace,
+          system: command.system,
+          status: command.status,
+          command: command.command,
+          name: generateCommandName(false, command.command),
+          action: (
+            <IconButton
+              size="small"
+              color="error"
+              onClick={() => delCommand(command.id)}
+              aria-label="delete"
+            >
+              <DeleteIcon />
+            </IconButton>
+          ),
+        }
+      }) ?? []
+    )
+  }, [blocked, deleteBlockList])
+
+  return (
+    <Box>
+      <Tooltip title="Add command">
+        <Button
+          style={{ float: 'right' }}
+          variant="contained"
+          color="primary"
+          aria-label="Add command"
+          onClick={() => setOpen(true)}
+        >
+          <AddIcon />
+        </Button>
+      </Tooltip>
+      <ModalWrapper
+        open={open}
+        header="Add Commands to Blocklist"
+        onClose={() => {
+          setOpen(false)
+          setSelected([])
+        }}
+        onCancel={() => {
+          setOpen(false)
+          setSelected([])
+        }}
+        onSubmit={() => {
+          addBlockList(selectedCommands)
+          setOpen(false)
+          setSelected([])
+        }}
+        content={
+          <Table
+            tableName=""
```

Comment:
Passing an empty name causes a minor problem. The `Table` implementation uses the name to create a local storage key to keep track of how many items to show, which columns to hide, etc. So if we were to add another table without a name there'd be a collision. A low-grief fix is to swap the text "Add Commands to Blocklist" out of the modal header and into the `tableName` variable here, i.e. kicking the can down the road.

A much better solution IMO is to refactor the table to take an optional prop that tells it what its local storage key should be and to fallback on the name if it isn't present. I'll leave it up to you whether you want to tackle that now or we can just open a bugfix issue later if you prefer.